scripts/testing/tolerance_test.py

```diff
@@ -1,78 +1,235 @@
+#!/bin/sh
+"exec" "python3" "-u" "-B" "$0" "$@"
+##############################################################################
+# Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+# other Serac Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+##############################################################################
+
+import argparse
 import json
-import argparse # for command line arguments
-import math # for comparing values within tolerance
-
- # l2norms last number in last temp listed changed from 9 to 8 (in small_change)
- # mins first number in second temp listed changed from 1 to 100 (big_change)
- 
- # load name of each field from file & populate them in a list 
-def get_field_names(d):    
+import math
+import os
+import sys
+
+
+# This script compares two Serac summary files against each other.
+# The 'baseline' file is treated as a truth against the possibly wrong
+# 'test' file. It attempts to fail with the most error messages it can at a given
+# correctness level (field names > sample names > sample values).
+# Summary files were the same within the given tolerance if the
+# script exists successfully and not otherwise.
+
+def ensure_file(path):
+    if not os.path.exists(path):
+        print("ERROR: Given file does not exist: {0}".format(path))
+        sys.exit(1)
+    if not os.path.isfile(path):
+        print("ERROR: Given file is not a file: {0}".format(path))
+        sys.exit(1)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Compare two Serac summary files")
+
+    parser.add_argument("--baseline", type=str, required=True,
+                        help="Path to baseline summary file")
+    parser.add_argument("--test", type=str, required=True,
+                        help="Path to test summary file")
+    parser.add_argument("--tolerance", type=float, required=True,
```

Comment:
It would be nice if we could specify this on a per-field basis, but that could be a follow-on PR.

Reply:
#601